### PR TITLE
feat(ramp): add bottom sheet quotes

### PR DESCRIPTION
**Pull request by wachunei**
## **Description**

This PR introduces a new presentation of the Quotes view for both Buy and Sell flows. The new presentation is in a modal style that reduces the amount of quotes displayed to the top 2 quotes. In case the available quotes are more than 2, the user can expand the [Bottom Sheet](https://www.notion.so/metamask-consensys/Bottom-Sheet-302164aff8a74b6eb721f49586f88ecd) component and view all quotes.

## **Related issues**

Fixes: https://consensyssoftware.atlassian.net/browse/ONRAMP-1338

## **Manual testing steps**

1. Go to Buy or Sell flow
2. Get to build quote view and press Get Quotes.

### **Before**

https://github.com/MetaMask/metamask-mobile/assets/1024246/211e6c03-9e60-4ca7-bfe0-7ab6f61fda7d

### **After**

https://github.com/MetaMask/metamask-mobile/assets/1024246/737beffb-b7cb-4e04-a0f4-f22d97cc9251

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/UI/Ramp/components/ErrorView.tsx [hunk 1]
```diff
@@ -51,6 +56,7 @@ interface Props {
   ctaLabel?: string; // The CTA button label, default will be "Try again" (Optional)
   ctaOnPress?: () => any; // The optional callback to be invoked when pressing the CTA button (Optional)
   icon?: IconType;
+  asScreen?: boolean; // Wether this component should be rendered as a screen or not (Optional)
```
Comment (Cal-L) [inline app/components/UI/Ramp/components/ErrorView.tsx:59]:
```suggestion
  asScreen?: boolean; // Whether this component should be rendered as a screen or not (Optional)
```

just small typo

**Review by Cal-L — COMMENTED** [review]
Just 1 small typo. Looks like a smoke test is failing - is it related to this PR?

**Comment by wachunei** [comment]
> Looks like a smoke test is failing - is it related to this PR?
@Cal-L I don't think it is related. Changes are within the Ramp folder only

**Review by Cal-L — DISMISSED** [review]
LGTM

**Comment by wachunei** [comment]
Currently blocked by:
- https://github.com/MetaMask/metamask-mobile/pull/8752
- https://github.com/MetaMask/metamask-mobile/pull/8755

**Comment by bkirb** [comment]
@wachunei LGTM I finished verifying on several device sizes ✅

**Review by Cal-L — APPROVED** [review]
Lgtm

**Pull request merged**